### PR TITLE
Add observability ready util

### PR DESCRIPTION
**Pull request by mayurkale22**
- Add an experimental artifact `opencensus-contrib-observability-ready-util`, that allows users to use OpenCensus easily.

- This will simplify all the OpenCensus initialization steps.

### CODE REVIEW

File: contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    // set sampling rate
+    TraceConfig traceConfig = Tracing.getTraceConfig();
+    TraceParams activeTraceParams = traceConfig.getActiveTraceParams();
+    traceConfig.updateActiveTraceParams(
+        activeTraceParams.toBuilder().setSampler(Samplers.probabilitySampler(probability)).build());
+
+    // create and register Trace Agent Exporter
+    OcAgentTraceExporter.createAndRegister(
+        OcAgentTraceExporterConfiguration.builder()
```
Comment (bogdandrutu) [inline contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: What about we add this in the ocagent artifact?

File: contrib/observability_ready_util/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# OpenCensus Observability Ready Util for Java
+
+[![Build Status][travis-image]][travis-url]
+[![Windows Build Status][appveyor-image]][appveyor-url]
+
+The *OpenCensus Observability Ready Util for Java* that allows users to use OpenCensus easily.
```
Comment (rghetia) [inline contrib/observability_ready_util/README.md]: remove 'that'

File: contrib/observability_ready_util/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+By default:
```
Comment (rghetia) [inline contrib/observability_ready_util/README.md]: Remove 'By default' and simply say 
```
It provides a wrapper that
- Enables Basic RPC Views
- Creates and Registers OCAgent Exporter to collect traces and metrics.
```
'By Defaults' suggest that there is a way to override the default but there isn't.

File: contrib/observability_ready_util/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  </dependency>
+</dependencies>
+```
+
+For Gradle add to your dependencies:
+```groovy
+compile 'io.opencensus:opencensus-contrib-observability-ready-util:0.25.0'
+```
+
+### The following code:
```
Comment (rghetia) [inline contrib/observability_ready_util/README.md]: may be change the title to 'Initialize OpenCensus' or 'Enable OpenCensus'

File: contrib/observability_ready_util/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+```java
+public class YourClass {
```
Comment (rghetia) [inline contrib/observability_ready_util/README.md]: add import statement.

File: contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  /**
+   * Enables OpenCensus metric and traces.
+   *
+   * <p>This will register all basic {@link io.opencensus.stats.View}s. When coupled with an agent,
+   * it allows users to monitor application behavior.
+   *
+   * <p>Please note that in addition to calling this method, the application must:
+   *
+   * <ul>
+   *   <li>Include opencensus-contrib-observability-ready-util dependency on the classpath
```
Comment (rghetia) [inline contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: is it necessary to mention this?

File: contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    TraceParams activeTraceParams = traceConfig.getActiveTraceParams();
+    traceConfig.updateActiveTraceParams(
+        activeTraceParams.toBuilder().setSampler(Samplers.probabilitySampler(probability)).build());
+
+    // create and register Trace Agent Exporter
+    OcAgentTraceExporter.createAndRegister(
+        OcAgentTraceExporterConfiguration.builder()
+            .setEndPoint(endPoint)
+            .setServiceName(serviceName)
+            .setUseInsecure(true)
```
Comment (rghetia) [inline contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: For production deployment Insecure will not work in most if not all cases.

File: contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java (hunk 1 above)
Comment (mayurkale22) [reply contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: As discussed, will open separate PR to show how this basic setup looks like in the exporter. 
Btw, ocagent has multiple artifacts, [contrib-agent](https://github.com/census-instrumentation/opencensus-java/tree/master/contrib/agent) artifact and ocagent [exporter](https://github.com/census-instrumentation/opencensus-java/tree/master/exporters/trace/ocagent). I think you are referring to exporter one, can you pls confirm.

Comment (mayurkale22) [reply contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: Alternative approach: https://github.com/census-instrumentation/opencensus-java/pull/2003

Comment (mayurkale22) [reply contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: I had a discussion with @rghetia and @dinooliva, we think this `opencensus-contrib-observability-ready-util` artifact is much more meaningful and useful than the exporter wise basic setup function.

1. Exporter level function would create confusion the users.
2. Exporter level function is just reducing 1-2 line of code i.e. enabling the views or sampling rate.
3. It looks weird to enable views using exporter level setup function.
4. We can start with this util and add same thing for SD or other exporters in the future, based on the adoption and usage pattern.
5. Mostly this `opencensus-contrib-observability-ready-util` will be used by Single click deployment, so makes sense to have agent exporter by default.

@rghetia and @dinooliva please add more, if anything is incorrect or missing.

@bogdandrutu WDYT?

File: contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java (hunk 7 above)
Comment (mayurkale22) [reply contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: Is it ok to assign it `false` (by default) or provide configurable param to control that, What do you suggest?

**Comment by mayurkale22** [comment]
@rghetia I have fixed the comments, please review again when you get a chance.

File: contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+   * </dependency>
+   * }</pre>
+   *
+   * <p>It is recommended to call this method before doing any RPC call to avoid missing stats.
+   *
+   * <pre>{@code
+   * BasicSetup.enableOpenCensus("with-service-name");
+   * }</pre>
+   *
+   * @param serviceName the service name.
```
Comment (rghetia) [inline contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: what is the service name and what is it used for?

Comment (mayurkale22) [reply contrib/observability_ready_util/src/main/java/io/opencensus/contrib/observability/ready/util/BasicSetup.java]: The service name specifies the name used for tracing/metrics. Used by Jaeger and Zipkin exporters, maybe to group the spans together.

**Comment by mayurkale22** [comment]
@dinooliva It would be nice if you can review and approve the PR.

**Comment by mayurkale22** [comment]
@rghetia Is it good to merge?

**Pull request merged**